src/System.Net.Ping/tests/FunctionalTests/PingTest.cs

```diff
@@ -32,6 +34,19 @@ protected override void Dispose(bool disposing)
             }
         }
 
+        private static void PingResultValidator(PingReply pingReply, IPAddress localIpAddress)
+        {
+            if (pingReply.Status == IPStatus.TimedOut)
+            {
+                // Workaround OSX ping6 bug, refer issue #15018
+                Assert.Equal(AddressFamily.InterNetworkV6, localIpAddress.AddressFamily);
```

Comment:
This means we see all the failures on IPv6 ? fine.

Reply:
This bug is expected only from ping6, hence the check.